src/OpenTelemetry/README.md

```diff
@@ -84,6 +84,44 @@ using var otel = Sdk.CreateTracerProvider(b => b
   * [Building your own Processor](../../docs/trace/extending-the-sdk/README.md#processor)
   * [Building your own Sampler](../../docs/trace/extending-the-sdk/README.md#sampler)
 
+### Troubleshooting
+
+Self diagnostics module is provided to help troubleshooting.
+
+To enable self diagnostics, go to the current directory of your process and
+create a configuration file named `DiagnosticsConfiguration.json` with the
+following content:
+
+```json
+{
+    "LogDirectory": ".",
+    "FileSize": 1024,
+    "EventLevel": "Error"
+}
+```
+
+`LogDirectory` is the directory of the output file. It can be an absolute path
+or a relative path to the current directory. `FileSize` is a positive integer,
+which specifies the log file size in
+[KiB](https://en.wikipedia.org/wiki/Kibibyte).
+[`EventLevel`](https://docs.microsoft.com/dotnet/api/system.diagnostics.tracing.eventlevel)
+is the level of the events to be captured.
+
+The SDK will attempt to open the configuration file in non-exclusive read-only
+mode, read the file and parse it as the configuration file every 3 seconds. If
+the SDK fails to parse the `LogDirectory` or `FileSize` fields as the specified
```

Comment:
What about `EventLevel`?

Reply:
Added `EventLevel` in-line. It has to be present and of valid value.